web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -285,53 +287,66 @@ private void connectToWorker() {
-                                    item.suppressEvents();
-                                });
-                            })
-                            .filter(cts -> !cts.isEmpty())
-                            .peek(cts -> {
-                                cts.forActiveTables(t -> {
-                                    assert t.state().isAncestor(cts)
-                                            : "Invalid binding " + t + " (" + t.state() + ") does not contain " + cts;
-                                });
-                            })
-                            .toArray(ClientTableState[]::new);
-                    // clear caches
-                    List<ClientTableState> inactiveStatesToRemove = cache.getAllStates().stream()
-                            .filter(ClientTableState::isEmpty)
-                            .collect(Collectors.toList());
-                    inactiveStatesToRemove.forEach(cache::release);
-
-                    flushable.clear();
+                    if (newSession) {
+                        // assert false : "Can't yet rebuild connections with new auth, please log in again";
+                        // nuke pending callbacks, we'll remake them
+                        handleCallbacks = new JsWeakMap<>();
+                        definitionCallbacks = new JsWeakMap<>();
+
+                        // for each cts in the cache, get all with active subs
+                        ClientTableState[] hasActiveSubs = cache.getAllStates().stream()
+                                .peek(cts -> {
+                                    cts.getHandle().setConnected(false);
+                                    cts.setSubscribed(false);
+                                    cts.forActiveLifecycles(item -> {
+                                        assert !(item instanceof JsTable) ||
+                                                ((JsTable) item).state() == cts
+                                                : "Invalid table state " + item + " does not point to state " + cts;
+                                        item.suppressEvents();
+                                    });
+                                })
+                                .filter(cts -> !cts.isEmpty())
+                                .peek(cts -> {
+                                    cts.forActiveTables(t -> {
+                                        assert t.state().isAncestor(cts)
+                                                : "Invalid binding " + t + " (" + t.state() + ") does not contain "
+                                                        + cts;
+                                    });
+                                })
+                                .toArray(ClientTableState[]::new);
+                        // clear caches
+                        // TODO verify we still need to do this, it seems like it might signify a leak
```

Comment:
does this need a deephaven-core issue number?

Reply:
Yes, or I should just finish fixing this.